certs-manager/src/main/java/io/strimzi/certs/SecretCertProvider.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Provides Kubernetes secrets containing certificates
+ */
+public class SecretCertProvider {
```

Comment:
I think this will need to be a bit more sophisticated ... We will probably need some secret / CM with the CA public key only. Maybe the CA public key should be also in the secret with the signed key. Maybe a Fluent API might be helpful here?

Reply:
Agree. It's part of the iterative process I'd like to use. I'll start use that and improving overtime based on needs we have.